import-scripts/study_directory_uses_updated_filenames.py

```diff
@@ -0,0 +1,538 @@
+
+def parse_current_datatypes(cancer_study_identifier):
+    """
+    the contents of the datatype sheet (in tab separated format) are parsed and all filenames (both meta and staging) or wildcard patterns are mapped to their associated datatypes. The maps are returned.
+    """
+    datatype_sheet_lines = datatype_sheet_download_tab_separated().splitlines()
+    filename2datatype = {}
+    wildcard_pattern2datatype = {}
+    header_skipped = False
+    for line in datatype_sheet_lines:
+        if not header_skipped:
+            if line.startswith("datatype"):
+                header_skipped = True
+            continue
+        fields = line.split("\t")
+        datatype = fields[0].strip()
+        staging_filename = fields[5].replace(CANCER_STUDY_TAG(), cancer_study_identifier).strip()
+        meta_filename = fields[9].replace(CANCER_STUDY_TAG(), cancer_study_identifier).strip()
+        if "*" in staging_filename:
+            wildcard_pattern2datatype[staging_filename] = datatype
+        else:
+            filename2datatype[staging_filename] = datatype
+        if "*" in meta_filename:
+            wildcard_pattern2datatype[meta_filename] = datatype
+        else:
+            filename2datatype[meta_filename] = datatype
+    return filename2datatype, wildcard_pattern2datatype
+
+def is_normal_sample_filename(filename):
+    """
+    examines filename to see if the importer would skip it as a "normal sample" file.
+    """
+    normal_sample_suffixes = ["_normals.txt", "data_mrna_seq_v2_rsem_normal_samples.txt", "data_mrna_seq_v2_rsem_normal_samples_zscores_ref_normal_samples.txt"]
+    for suffix in normal_sample_suffixes:
+        if (filename.endswith(suffix)):
+            return True;
+    return False;
+
+def determine_cancer_study_identifier_from_meta_file(filepath):
+    """
+    search file at <filepath> for a cancer_study_identifier and return it if found
+    """
+    CANCER_STUDY_IDENTIFIER_PREFIX = "cancer_study_identifier:"
+    file = open(filepath, 'r')
+    for line in file.readlines():
+        line_without_whitespace = "".join(line.split()).strip()
+        if line.startswith(CANCER_STUDY_IDENTIFIER_PREFIX):
+            cancer_study_identifier = line[len(CANCER_STUDY_IDENTIFIER_PREFIX):].strip()
+            if len(cancer_study_identifier) > 0:
+                return cancer_study_identifier 
+    return None
+
+def determine_cancer_study_identifier(basedir, filenames):
+    """
+    search <filenames> in <basedir> for a cancer_study_identifier in any meta filename and return it if found
+    """
+    cancer_study_identifier = None
+    for filename in filenames:
+        if filename.startswith("meta_"):
+            filepath = os.path.join(basedir, filename)
+            if os.path.isfile(filepath):
+                cancer_study_identifier = determine_cancer_study_identifier_from_meta_file(filepath)
+        if cancer_study_identifier is not None:
```

Comment:
maybe break the loop once it's set and return cacner_study_identifier at the end of the function

Reply:
done